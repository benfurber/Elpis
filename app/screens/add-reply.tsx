import React, { Component } from "react";
import { StyleSheet, TextInput } from "react-native";

import { Mutation } from "react-apollo";
import { withMappedNavigationParams } from "react-navigation-props-mapper";

import { Comment, NavigationType } from "interfaces";
import { labels } from "labels";
import { ADD_REPLY } from "mutations";
import { FormContainerScreen } from "screens";
import { layout, typography, colours } from "styles";

interface Props {
  commentId: Comment["id"];
  navigation: NavigationType;
}

interface State {
  textInput: string;
  textInputEditable: boolean;
}

class AddReplyScreen extends Component<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      textInputEditable: true,
      textInput: "",
    };
  }

  onSubmitEditing(query) {
    const { textInput } = this.state;
    const { commentId, navigation } = this.props;

    this.setState({ textInputEditable: false });

    query({
      variables: {
        content: textInput,
        id: commentId,
      },
    }).then(() => {
      this.setState({ textInput: "", textInputEditable: true });
      navigation.dismiss();
    });
  }

  renderAddResponse(mutation) {
    return (
      <Mutation mutation={mutation}>
        {(createReply, {}) => (
          <TextInput
            onSubmitEditing={() => this.onSubmitEditing(createReply)}
            style={styles.text}
            autoFocus={true}
            editable={this.state.textInputEditable}
            multiline={true}
            placeholder={labels.addPlaceholderTitle}
            onChangeText={textInput => this.setState({ textInput })}
            value={this.state.textInput}
            returnKeyLabel={labels.submit}
          />
        )}
      </Mutation>
    );
  }

  render() {
    const { commentId, navigation } = this.props;

    return (
      <FormContainerScreen
        analyticsContent={{ contentType: "AddReply", contentId: commentId }}
        navigation={navigation}
        title={labels.addYourReply}
      >
        {this.renderAddResponse(ADD_REPLY)}
      </FormContainerScreen>
    );
  }
}

const styles = StyleSheet.create({
  label: {
    color: colours.mediumGrey,
    fontWeight: "bold",
    paddingTop: layout.spacingXL,
    paddingHorizontal: layout.spacingL,
  },
  textInputContainer: {
    width: "100%",
    flexDirection: "column",
  },
  text: {
    fontFamily: "creteround-regular",
    fontSize: typography.fontSizeL,
    padding: layout.spacingL,
    width: "100%",
  },
});

const wrappedAddReplyScreen = withMappedNavigationParams()(AddReplyScreen);
export {
  wrappedAddReplyScreen as AddReplyScreen,
  AddReplyScreen as UnwrappedAddReplyScreen,
};
