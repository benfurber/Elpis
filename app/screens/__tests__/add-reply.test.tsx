import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { UnwrappedAddReplyScreen } from "../add-reply";

let navigation;
jest.mock(navigation, () => jest.fn());

describe("AddReplyScreen", () => {
  it("renders correctly", () => {
    const component = shallow(
      <UnwrappedAddReplyScreen navigation={navigation} commentId={"7sdfbu"} />,
    );

    expect(component).toMatchSnapshot();
  });
});
