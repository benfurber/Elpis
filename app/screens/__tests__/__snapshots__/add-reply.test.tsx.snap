// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReplyScreen renders correctly 1`] = `
<FormContainerScreen
  analyticsContent={
    Object {
      "contentId": "7sdfbu",
      "contentType": "AddReply",
    }
  }
  title="Adicione a sua resposta"
>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "createReply",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "imagePath",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "imagePath",
                        },
                      },
                    },
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "content",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "content",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "createReply",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "ID",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "URL",
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "imagePath",
                  },
                },
              },
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NonNullType",
                  "type": Object {
                    "kind": "NamedType",
                    "name": Object {
                      "kind": "Name",
                      "value": "String",
                    },
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "content",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 160,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Mutation>
</FormContainerScreen>
`;
